apps/meteor/ee/app/livechat-enterprise/server/api/lib/contacts.ts

```diff
@@ -0,0 +1,52 @@
+import type { ILivechatContact, IUser } from '@rocket.chat/core-typings';
+import { License } from '@rocket.chat/license';
+import { LivechatContacts, LivechatRooms, LivechatVisitors } from '@rocket.chat/models';
+
+import { Livechat } from '../../../../../../app/livechat/server/lib/LivechatTyped';
+import { i18n } from '../../../../../../server/lib/i18n';
+
+export async function changeContactBlockStatus({ contactId, block, visitorId }: { contactId: string; visitorId: string; block: boolean }) {
```

Comment:
Some units would be great for this func

Reply:
Isn't the e2e enough? I mean, there's nothing too crazy that we're doing here that isn't being validated there. Most of these 'ifs' are just to ensure since a contact will never have a channel without a visitor and vice-versa

Reply:
Think that's fair